src/sql/expr.rs

```diff
@@ -1394,6 +1419,18 @@ impl ScalarExpr {
     }
 }
 
+fn negate(f: &BinaryFunc) -> Option<BinaryFunc> {
```

Comment:
I'd have written this as a method on `BinaryFunc`, but if you prefer a free function no worries. Maybe that's idiomatic for the file (a bunch of other builder idioms, though).